src/coord/src/catalog/storage.rs

```diff
@@ -481,6 +517,36 @@ impl Transaction<'_> {
         }
     }
 
+    pub fn insert_timestamp(
+        &self,
+        source_id: &GlobalId,
+        partition_id: &str,
+        timestamp: Timestamp,
+        offset: i64,
+    ) -> Result<(), Error> {
+        match self
+            .inner
+            .prepare_cached(
+                "INSERT OR IGNORE INTO timestamps (sid, pid, timestamp, offset) VALUES (?, ?, ?, ?)",
+            )?
+            .execute(params![SqlVal(source_id), partition_id, timestamp, offset])
```

Comment:
is there a better way to insert a list of multiple timestamps with just one sql statement?

Reply:
based on the docs it seems multiple sql statements within a txn is fine